.github/workflows/test.yml

```diff
@@ -741,14 +767,21 @@ jobs:
         path: gomodcache
         key: ${{ needs.gomod-cache.outputs.cache-key }}
         enableCrossOsArchive: true
-    - name: run staticcheck
+    - name: run staticcheck (${{ matrix.name }})
       working-directory: src
       run: |
         export GOROOT=$(./tool/go env GOROOT)
         ./tool/go run -exec \
           "env GOOS=${{ matrix.goos }} GOARCH=${{ matrix.goarch }}" \
            honnef.co/go/tools/cmd/staticcheck -- \
-           $(env GOOS=${{ matrix.goos }} GOARCH=${{ matrix.goarch }} ./tool/go list ./... | grep -v tempfork)
+           $(./tool/go run ./tool/listpkgs --ignore-3p --goos=${{ matrix.goos }} --goarch=${{ matrix.goarch }} ${{ matrix.flags }} ./...)
+
+  staticcheck:
+    needs: staticcheck-shard
+    runs-on: ubuntu-24.04
+    if: ${{ needs.test.result == 'success' }}
+    steps:
+      - run: echo "all staticcheck variants passed"
```

Comment:
Why do we need this - isn't it going to be clear that all variants passed because the job did not fail?